go/vt/vtgate/planbuilder/operators/sharded_routing.go

```diff
@@ -486,7 +486,7 @@ func (tr *ShardedRouting) processMultiColumnVindex(
 		return newVindexFound
 	}
 
-	var newOption []*VindexOption
+	newOption := make([]*VindexOption, 0, len(v.Options))
```

Comment:
I don't know about this specific case, but in general, is it really correct to allocate it all everywhere? Do we not have cases where it's more the exception an element is added and not the common case? And we very much over-allocate? 

That's a general issue with this linter, it's not very smart and complains about every case, while not every case makes really sense to convert.

Reply:
> I don't know about this specific case, but in general, is it really correct to allocate it all everywhere?

@dbussink I haven't validated it myself, but I've consistently read _(and just confirmed with ChatGPG)_ that allocating a slice with a fixed capacity is "usually" more efficient, because a zero-capacity slice has to be grown every append, causing new allocations for a new backing array each append

To my knowledge the risk to using a capacity that may/may not be 100% used is memory usage that may just be filled with empty values. But there must be cases where this isn't always true, perhaps really huge slices(?). I wondered if you had any scenarios in mind?